src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/AssessmentTest.kt

```diff
@@ -477,6 +481,464 @@ class AssessmentTest : IntegrationTestBase() {
     }
   }
 
+  @Test
+  fun `Get all cas3 assessments correctly without any query parameter `() {
+    `Given a User` { user, jwt ->
+      `Given Some Offenders` { offenderSequence ->
+        val applicationSchema = temporaryAccommodationApplicationJsonSchemaEntityFactory.produceAndPersist {
+          withPermissiveSchema()
+        }
+
+        val assessmentSchema = temporaryAccommodationAssessmentJsonSchemaEntityFactory.produceAndPersist {
+          withPermissiveSchema()
+          withAddedAt(OffsetDateTime.now())
+        }
+
+        val (offender, otherOffender) = offenderSequence.take(2).toList()
+
+        val application = temporaryAccommodationApplicationEntityFactory.produceAndPersist {
+          withCrn(offender.first.otherIds.crn)
+          withCreatedByUser(user)
+          withApplicationSchema(applicationSchema)
+          withProbationRegion(user.probationRegion)
+        }
+
+        val otherApplication = temporaryAccommodationApplicationEntityFactory.produceAndPersist {
+          withCrn(otherOffender.first.otherIds.crn)
+          withCreatedByUser(user)
+          withApplicationSchema(applicationSchema)
+          withProbationRegion(user.probationRegion)
+        }
+
+        buildAndSaveAssessment(user, application, AssessmentDecision.ACCEPTED, assessmentSchema, null)
```

Comment:
Since its a integration test, i would like to keep the set of record consistent for all tests so that the 'given' data is same for all test and specific expected results. In this way, am not manipulating test data but make sure the results are correct in all situation.

Its local integration test db and would like to keep the data as realistic as possible.

Anyway, i have removed where its not needed, but i would prefer the data setup is same for all testcases.

Reply:
That's reasonable for improved clarity between tests, I hasn't considered that.